internal/app/handle.go

```diff
@@ -85,3 +113,77 @@ func handleGokuDefault(w http.ResponseWriter, r *http.Request) {
 	// Writes the JSON bytes to the JSON response writer
 	w.Write(d)
 }
+
+// lenOfOne handles what JSON message to give back for a JSON array
+// with one fighter in it
+func lenOfOne(f fighter, a *allegiance) {
+	if f.Name == "" {
+		logger.Error().Msg("Some fighter name JSON fields not specified, or empty.")
+		a.Message = "Some fighter name JSON fields not specified, or empty. Check the console log and the README."
+		return
+	}
+	if f.Power == 0 {
+		logger.Error().Msg("Fighter power JSON field not specified, or 0.")
+		a.Message = "Fighter power JSON field not specified, or 0. Check the console log and the README."
+		return
+	} else if f.Power > 9000 {
+		a.Message = "The scouter says " + f.Name + "'s power level is over 9000! You better start running."
+	} else if f.Power < 206 {
+		a.Message = f.Name + " is weaker than Krillin, cmon bruh."
+	} else {
+		a.Message = f.Name + "'s power level isn't even over 9000, they're a straight side character."
+	}
+}
+
+// lenOfTwo handles what JSON message to give back for a JSON array
+// with two fighters in it
+func lenOfTwo(f fighter, f1 fighter, a *allegiance) {
+	if f.Name == "" || f1.Name == "" {
+		logger.Error().Msg("Some fighter name JSON fields not specified, or empty.")
+		a.Message = "Some fighter name JSON fields not specified, or empty. Check the console log and the README."
+		return
+	}
+	if f.Power == 0 || f1.Power == 0 {
+		logger.Error().Msg("Some fighter power JSON fields not specified, or 0.")
+		a.Message = "Some fighter power JSON fields not specified, or 0. Check the console log and the README."
+		return
+	}
+	if f.Power == f1.Power && f.Power < 206 && f1.Power < 206 {
+		a.Message = f.Name + " and " + f1.Name + " are equally trash, they better fuse or something."
+	} else if f.Power < 206 && f1.Power < 206 {
+		a.Message = f.Name + " and " + f1.Name + " are both weaker than Krillin, im done."
+	} else if f.Power > f1.Power {
+		a.Message = f.Name + "'s power level is superior to " + f1.Name + "'s"
+	} else if f.Power == f1.Power {
+		a.Message = f.Name + "'s power level is equal to " + f1.Name + "'s"
+	} else {
+		a.Message = f1.Name + "'s power level is superior to " + f.Name + "'s"
+	}
+}
+
+//lenOverTwo handles what JSON message to give back for a JSON array
+// with more than two fighters in it
+func lenOverTwo(fs fighters, a *allegiance) {
+	maxPower := 0
+	maxFighter := fs.Fighters[0]
+	// loops through each fighter to compare their powers
+	for _, fighter := range fs.Fighters {
+		if fighter.Name == "" {
```

Comment:
the request validation shouldn't be happening here. this should be done as soon as the json is unmarshaled.